coderd/notifications/manager_test.go

```diff
@@ -0,0 +1,227 @@
+package notifications_test
+
+import (
+	"context"
+	"encoding/json"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/xerrors"
+
+	"cdr.dev/slog"
+	"cdr.dev/slog/sloggers/slogtest"
+
+	"github.com/coder/coder/v2/coderd/coderdtest"
+	"github.com/coder/coder/v2/coderd/database"
+	"github.com/coder/coder/v2/coderd/database/dbmem"
+	"github.com/coder/coder/v2/coderd/database/dbtestutil"
+	"github.com/coder/coder/v2/coderd/notifications"
+	"github.com/coder/coder/v2/coderd/notifications/dispatch"
+	"github.com/coder/coder/v2/coderd/notifications/types"
+	"github.com/coder/coder/v2/testutil"
+)
+
+func TestBufferedUpdates(t *testing.T) {
+	t.Parallel()
+
+	// setup
+	if !dbtestutil.WillUsePostgres() {
+		t.Skip("This test requires postgres")
+	}
+	ctx, logger, db, ps := setup(t)
+	interceptor := &bulkUpdateInterceptor{Store: db}
+
+	santa := &santaHandler{}
+	cfg := defaultNotificationsConfig(database.NotificationMethodSmtp)
+	mgr, err := notifications.NewManager(cfg, interceptor, logger.Named("notifications-manager"))
+	require.NoError(t, err)
+	mgr.WithHandlers(map[database.NotificationMethod]notifications.Handler{
+		database.NotificationMethodSmtp: santa,
+	})
+	enq, err := notifications.NewStoreEnqueuer(cfg, interceptor, defaultHelpers(), logger.Named("notifications-enqueuer"))
+	require.NoError(t, err)
+
+	client := coderdtest.New(t, &coderdtest.Options{Database: db, Pubsub: ps})
+	user := coderdtest.CreateFirstUser(t, client)
```

Comment:
It's too heavyweight to create a whole Coderd just to make a user.  You want `dbgen.User()`